schema.go

```diff
@@ -1,8 +1,72 @@
 package kallax
 
+import "fmt"
+
+// Schema represents a table schema in the database. Contains some information
+// like the table name, its columns, its identifier and so on.
 type Schema interface {
-	Alias() string
-	Table() string
-	Identifier() string
-	Columns() []string
+	// GetAlias returns the name of the alias used in queries for this schema.
+	GetAlias() string
```

Comment:
Then you'd have to make a constructor with a bazillion arguments instead of just constructing the type.

Also, the type that implements `Schema` is embedded in the generated schema, which can override the method by having an `Alias` field in the model.

```go
type Foo struct {
  kallax.Model
  Alias string
}
```

would generate a schema like

```
type fooSchema struct {
  *kallax.BaseSchema
  Alias kallax.SchemaField
}
```

And the Alias field has preference over the method when calling `schema.Alias()`

The schema methods is not something you use manually, they are only used internally by the generated code, so that's why we preferred to add `Get` before the method name and avoid possible name conflicts.